python/pycylon/test/test_data_split.py

```diff
@@ -61,4 +61,8 @@ def test_data_split():
     env.barrier()
     
     if env.rank == 0:
-        os.remove(data_file)
+        try:
+            os.remove(data_file)
+            return False
+        except OSError:    # file is in use
+            return True
```

Comment:
I assume you're trying to remove the data after doing the test? If that is the case, it is better to leverage `pytest` and use a temporary directory which gets cleaned up by the test framework itself. Looking into the exising code, previous please can also be cleaned up. 

https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html

Reply:
os.remove(data_file) was used before and caused an intermittent error during running on a windows machine. 

I just added the code with exception handling. This issue is happening only in this test case. So, we should not make it standard for all other test cases.